src/main/java/com/matag/game/turn/action/combat/DeclareBlockerService.java

```diff
@@ -29,7 +30,15 @@ public void declareBlockers(GameStatus gameStatus, Map<Integer, List<Integer>> b
 
     blockingCreaturesIdsForAttackingCreaturesIds.forEach((blockingCreatureId, blockedCreaturesIds) -> {
       CardInstance blockedCreature = currentPlayer.getBattlefield().findCardById(blockedCreaturesIds.get(0));
-      nonCurrentPlayer.getBattlefield().findCardById(blockingCreatureId).checkIfCanBlock(blockedCreature);
+      CardInstance mainBlocker = nonCurrentPlayer.getBattlefield().findCardById(blockingCreatureId);
+      List<CardInstance> blockers = blockingCreaturesIdsForAttackingCreaturesIds
```

Comment:
Oh, I got it now.

Much better to extract into a service.

Reply:
Well, got it... needs to be like this.

Reply:
I like how you basically ask a question and answer it yourself, it makes my job so much easier 👍